helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/PerInstanceAccessor.java

```diff
@@ -308,20 +309,34 @@ public Response updateInstanceConfig(@PathParam("clusterId") String clusterId,
     }
     InstanceConfig instanceConfig = new InstanceConfig(record);
     ConfigAccessor configAccessor = getConfigAccessor();
+
     try {
+      /*
+       * Even if the instance is disabled, non-valid instance topology config will cause rebalance
+       * failure. We are doing the check whenever user updates InstanceConfig.
+       */
+      if (command == Command.delete || command == Command.update) {
+        validateDeltaTopologySettingInInstanceConfig(clusterId, instanceName, configAccessor,
```

Comment:
Why we dont call this in the switch case? It will make the code more clear. By doing that, you can let this validateDeltaTopologySettingInInstanceConfig, return a Config if it is valid. Otherwise, return null. 

So the following code will based on the return result to do the write to ZK.